vcd/resource_vcd_vapp_network.go

```diff
@@ -183,24 +217,41 @@ func resourceVcdVappNetworkCreate(d *schema.ResourceData, meta interface{}) erro
 					EndAddress: data["end_address"].(string)}}
 		}
 	}
-
-	task, err := vapp.AddIsolatedNetwork(vappNetworkSettings)
-
+	var orgVdcNetwork *types.OrgVDCNetwork
+	if networkId, ok := d.GetOk("org_network"); ok {
+		orgNetwork, err := vdc.GetOrgVdcNetworkByNameOrId(networkId.(string), true)
+		if err != nil {
+			return err
+		}
+		orgVdcNetwork = orgNetwork.OrgVDCNetwork
+	}
+	vAppNetworkConfig, err := vapp.AddNetwork(vappNetworkSettings, orgVdcNetwork)
 	if err != nil {
 		return fmt.Errorf("error creating vApp network. %#v", err)
 	}
 
-	err = task.WaitTaskCompletion()
-	if err != nil {
-		return fmt.Errorf("error waiting for task to complete: %+v", err)
+	vAppNetwork := types.VAppNetworkConfiguration{}
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == d.Get("name").(string) {
+			vAppNetwork = networkConfig
+		}
 	}
 
```

Comment:
Should it return an early error if `vAppNetwork` is actually empty after this lookup above to be more verbose?

Reply:
thank you. Missed in few places this check. Now added